tools/toolchains/iar.py

```diff
@@ -243,3 +243,10 @@ def make_ld_define(name, value):
     @staticmethod
     def redirect_symbol(source, sync, build_dir):
         return "--redirect %s=%s" % (source, sync)
+
+    @staticmethod
+    def redirect_main(dest, build_dir):
+        if dest != 'main':
+            return IAR.redirect_symbol("main", IAR.name_mangle(dest), build_dir)
+        else:
+            return None
```

Comment:
Since IAR's main function isn't wrapped like ARM and GCC_ARM no additional ld flags are needed when the name is main, so 'None' is returned.